components/scheduler2/common/util.go

```diff
@@ -0,0 +1,77 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	servercommon "go.temporal.io/server/common"
+	"go.temporal.io/server/components/scheduler2/core"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/hsm"
+)
+
+func ValidateTask[
+	S comparable,
+	SM hsm.StateMachine[S],
+	E any](node *hsm.Node, transition hsm.Transition[S, SM, E]) error {
+	if err := node.CheckRunning(); err != nil {
+		return err
+	}
+	sm, err := hsm.MachineData[SM](node)
+	if err != nil {
+		return err
+	}
+	if !transition.Possible(sm) {
+		return fmt.Errorf(
+			"%w: %w: cannot transition from state %v to %v",
+			consts.ErrStaleReference,
+			hsm.ErrInvalidTransition,
+			sm.State(),
+			transition.Destination,
+		)
+	}
+	return nil
+}
+
+// Intended to be called with a substate machine's node. Returns a cloned copy
```

Comment:
Use Go conventions for docstrings please.

```suggestion
// LoadSchedulerFromParent is intended to be called with a substate machine's node. Returns a cloned copy
```